src/overlays/actors/ovl_En_Mm2/z_en_mm2.c

```diff
@@ -31,16 +30,55 @@ const ActorInit En_Mm2_InitVars = {
     (ActorFunc)EnMm2_Draw,
 };
 
-#endif
+#include "overlays/ovl_En_Mm2/ovl_En_Mm2.c"
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Init.s")
+void EnMm2_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm2* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Destroy.s")
+    Actor_SetScale(&this->actor, 0.015f);
+    this->actionFunc = EnMm2_WaitForRead;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/func_809A2080.s")
+void EnMm2_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/func_809A20FC.s")
+// Action function whilst Link is reading the letter.
+void EnMm2_Reading(EnMm2* this, GlobalContext* globalCtx) {
+    u8 talkState = func_80152498(&globalCtx->msgCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Update.s")
+    if (talkState != 2) {
+        if (talkState == 5 && func_80147624(globalCtx)) {
+            func_801477B4(globalCtx);
+            this->actionFunc = EnMm2_WaitForRead;
+        }
+    } else {
+        this->actionFunc = EnMm2_WaitForRead;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Mm2/EnMm2_Draw.s")
+// Action function that awaits Link to read the letter, changing the A button to "Check" when he is within range to do
+// so (and facing the letter).
+void EnMm2_WaitForRead(EnMm2* this, GlobalContext* globalCtx) {
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        func_801518B0(globalCtx, 0x277B, &this->actor);
+        this->actionFunc = EnMm2_Reading;
+    } else if ((this->actor.xzDistToPlayer < 60.0f) && (Actor_IsLinkFacingActor(&this->actor, 0x3000, globalCtx))) {
+        func_800B8614(&this->actor, globalCtx, 110.0f);
+    }
+}
+
+void EnMm2_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnMm2* this = THIS;
+
+    this->actionFunc(this, globalCtx);
+}
+
+void EnMm2_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+
+    func_8012C28C(globalCtx->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), (0x00 | 0x02) | 0x00);
```

Comment:
```suggestion
    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
```
`gfxdis.f3dex2` gives the macros for these bits.